src/test/java/codeu/controller/ChatServletTest.java

```diff
@@ -510,21 +523,51 @@ public void testDoPut_SettingUsers() throws IOException, ServletException {
     HashSet<String> fakeUsersToBeAdded = new HashSet<>();
     fakeUsersToBeAdded.add("test_user1");
     fakeUsersToBeAdded.add("test_user2");
-    
+
     Mockito.when(mockConversation.getMembers()).thenReturn(fakeUsersToBeAdded);
 
     chatServlet.doPut(mockRequest, mockResponse);
 
     Mockito.verify(mockConversation).setMembers(fakeUsersToBeAdded);
     Mockito.verify(mockConversationStore).updateConversation(mockConversation);
+    Mockito.when(mockSession.getAttribute("subbed")).thenReturn(false);
+
+  }
+
+  @Test
+  public void testDoPut_RecievingNotificationsUnmute() throws IOException, ServletException {
+    UUID fakeConversationId = UUID.randomUUID();
+    Mockito.when(mockSession.getAttribute("user")).thenReturn("test_username");
+    Mockito.when(mockUserStore.getUser("test_username")).thenReturn(mockUser);
+    Mockito.when(mockRequest.getRequestURI()).thenReturn("/chat/" + fakeConversationId);
+    Mockito.when(mockConversationStore.getConversation(fakeConversationId)).thenReturn(mockConversation);
+    Mockito.when(mockRequest.getHeader("purpose")).thenReturn("recievingNotifications");
+    Mockito.when(mockReader.readLine()).thenReturn("unmute");
+
+    chatServlet.doPut(mockRequest, mockResponse);
+    //Mockito.verify(mockUser).addSubscription(fakeConversationId);
```

Comment:
By removing this line, the test doesn't actually test anything. Is this not working?

Reply:
Same for next test.